### PR TITLE
 JAMES-2399 Allow specifying per threshold templates

### PR DESCRIPTION
**Pull request by chibenwa**
Only the last commit

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please (Yoda issue)

**Comment by chibenwa** [comment]
test this please (build rescheduling)

File: mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/mailing/QuotaMailingListenerConfiguration.java [hunk 1]
```diff
@@ -179,20 +220,25 @@ public static QuotaMailingListenerConfiguration defaultConfiguration() {
...
         this.thresholds = thresholds;
         this.gracePeriod = gracePeriod;
-        this.bodyTemplate = bodyTemplate;
-        this.subjectTemplate = subjectTemplate;
         this.name = name;
     }
 
+    public RenderingInformation getRenderingInformation(QuotaThreshold quotaThreshold) {
+        return Optional.ofNullable(toRenderingInformation.get(quotaThreshold))
+            .orElse(DEFAULT_RENDERING_INFORMATION);
```
Comment (mbaechler) [inline mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/mailing/QuotaMailingListenerConfiguration.java]: you manage the default value twice, there's probably something wrong

Comment (chibenwa) [reply mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/mailing/QuotaMailingListenerConfiguration.java]: No. You can ommit rendering informations. Or you can omit a part of this information.

**Comment by chibenwa** [comment]
> I'm not sure it's what was requested. I'd rather allow both (global is the default, per threshold is for override).

Got it.

Seems easily doable...

**Comment by chibenwa** [comment]
```
[9e121648a70c0b920e48d56c212f55a50604aee6] Tests in error: 
[9e121648a70c0b920e48d56c212f55a50604aee6]   ToRepositoryTest.userShouldBeAbleToAccessReprocessedMails:130 » ConditionTimeout
```

test this please

**Comment by chibenwa** [comment]
```
[c40a4d358b417280855a3d6c59797154b123b328]   ManageableMailQueueContract.concurrentEnqueueShouldNotAlterBrowsing:245 
[c40a4d358b417280855a3d6c59797154b123b328] Actual and expected should have same size but actual size was:
```

test this please

**Comment by chibenwa** [comment]
Was :green_apple: 

Forced pushed to solve conflicts.

File: mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/mailing/QuotaMailingListenerConfiguration.java [hunk 2]
```diff
@@ -130,21 +207,21 @@ public Builder subjectTemplate(String subjectTemplate) {
...
-            return this;
-        }
-
         public Builder bodyTemplate(Optional<String> bodyTemplate) {
             bodyTemplate.ifPresent(this::bodyTemplate);
             return this;
         }
 
-        public Builder subjectTemplate(Optional<String> subjectTemplate) {
+        public Builder subjectTemplate(Optional<String>  subjectTemplate) {
```
Comment (aduprat) [inline mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/mailing/QuotaMailingListenerConfiguration.java]: extra space?

**Comment by mbaechler** [comment]
merged

**Pull request closed without merging**